pyccel/codegen/wrapper/cuda_to_c_wrapper.py

```diff
@@ -0,0 +1,96 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+Module describing the code-wrapping class : CudaToCWrapper
+which creates an interface exposing Cuda code to C.
+"""
+
+from pyccel.parser.scope import Scope
+from pyccel.ast.core import Import, FunctionDef
+from pyccel.ast.cwrapper      import PyModule
+from pyccel.ast.core import AsName
+from .wrapper import Wrapper
+
+
+class CudaToCWrapper(Wrapper):
+    """
+    Class for creating a wrapper exposing Fortran code to C.
+
+    A class which provides all necessary functions for wrapping different AST
+    objects such that the resulting AST is C-compatible. This new AST is
+    printed as an intermediary layer.
+    """
+    def __init__(self):
+        super().__init__()
+    
+    def _wrap_Module(self, expr):
+        """
+        Build a `PyModule` from a `Module`.
+        Create a `PyModule` which wraps a C-compatible `Module`.
+        Parameters
+        ----------
+        expr : Module
+            The module which can be called from C.
+        Returns
+        -------
+        PyModule
+            The module which can be called from Python.
+        """
+        # Define scope
+        scope = expr.scope
+        mod_scope = Scope(used_symbols = scope.local_used_symbols.copy(), original_symbols = scope.python_names.copy())
+        self.scope = mod_scope
+
+        imports = [self._wrap(i) for i in getattr(expr, 'original_module', expr).imports]
+        imports = [i for i in imports if i]
+
+        # Wrap classes
+        classes = [self._wrap(i) for i in expr.classes]
+
+        # Wrap functions
+        funcs_to_wrap = [f for f in expr.funcs if f not in (expr.init_func, expr.free_func)]
+        funcs_to_wrap = [f for f in funcs_to_wrap if not f.is_inline]
+
+        # Add any functions removed by the Fortran printer
+        removed_functions = getattr(expr, 'removed_functions', None)
+        if removed_functions:
+            funcs_to_wrap.extend(removed_functions)
+
+        funcs = [self._wrap(f) for f in funcs_to_wrap]
+
+        # Wrap interfaces
+        interfaces = [self._wrap(i) for i in expr.interfaces if not i.is_inline]
+
+        init_func = self._build_module_init_function(expr, imports)
```

Comment:
What will `_build_module_init_function` do? I don't think this function needs a special treatment